packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -114,13 +114,24 @@ class LinkViewController extends PlatformViewController {
   factory LinkViewController.fromParams(
     PlatformViewCreationParams params,
     BuildContext context,
-  ) {
-    final int viewId = params.id;
-    final LinkViewController controller = LinkViewController(viewId, context);
-    controller._initialize().then((_) {
+  ) =>
+      LinkViewController(params.id, context).._asyncInit(params);
+
+  Future<void> _asyncInit(PlatformViewCreationParams params) async {
+    try {
+      await _initialize();
+      if (context == null) {
```

Comment:
I suggest keeping `context` final and adding a new field `_isDisposed`:
```suggestion
      if (_isDisposed) {
```